src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1579,10 +1601,12 @@ namespace cryptonote
           m_block_queue.remove_spans(span_connection_id, start_height);
 
           const uint64_t current_blockchain_height = m_core.get_current_blockchain_height();
+          const boost::posix_time::time_duration dt = boost::posix_time::microsec_clock::universal_time() - start;
+          const double blocks_per_seconds = (((current_blockchain_height - previous_height) * 1e6) / dt.total_microseconds());
```

Comment:
Looks like this could underflow in the event of a reorg to a lower height. I suggest just moving this into the if right below